pontoon/base/models.py

```diff
@@ -2585,12 +2602,19 @@ def map_entities(cls, locale, entities, visible_entities=None):
             )
         )
 
+        if preferred_source_locale != '':
+            entities = entities.prefetch_alternative_originals(preferred_source_locale)
+
         for entity in entities:
             translation_array = []
+            string = entity.string
+            string_plural = entity.string_plural
 
-            if entity.string_plural == "":
+            if string_plural == "":
                 translation = entity.get_active_translation().serialize()
                 translation_array.append(translation)
+                if preferred_source_locale != '' and entity.alternative_originals:
+                    string = entity.alternative_originals[0].string
```

Comment:
Now to handle plurals, we should actually move this block below the `if string_plural == '' - else` block and then use the last available plural form for the pluralized string:
```python
if string_plural != '':
    string_plural = entity.alternative_originals[-1].string
```

Reply:
This works for all the files I have in my Test project except the `amo.po` file. It works when only changing `original` to the `alternative_original` but the plural form causes an error with the JSON. ("The response content is not JSON-compatible")

When I try to simulate the query in the shell I get an `AssertionError: Negative indexing is not supported.`

Reply:
Hm, I'm not able to reproduce your issues. 

What do you mean by "changing original to the alternative_original"?

I've made the following change:
```diff
diff --git a/pontoon/base/models.py b/pontoon/base/models.py
index 105a600a7..820741e54 100644
--- a/pontoon/base/models.py
+++ b/pontoon/base/models.py
@@ -2621,6 +2621,11 @@ class Entity(DirtyFieldsMixin, models.Model):
                     translation = entity.get_active_translation(plural_form).serialize()
                     translation_array.append(translation)
 
+            if preferred_source_locale != '' and entity.alternative_originals:
+                string = entity.alternative_originals[0].string
+                if string_plural != '':
+                    string_plural = entity.alternative_originals[-1].string
+
             entities_array.append({
                 'pk': entity.pk,
                 'original': string,
```

Reply:
Ahh, OK.  I had them set up as stand-alone `if` statements instead of nesting the plurals `if` statement.  It is working fine now ~ Thank you.